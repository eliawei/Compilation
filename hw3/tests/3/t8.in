// cast set to int
int foo(int a) {
  set[0..10] set1;
  return (int)set1 + a;
}

void main() {
    set[5..6] myset1;
    int a = foo((int)(myset1 + 100));
}
