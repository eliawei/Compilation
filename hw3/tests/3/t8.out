---end scope---
a INT -1
set1 SET 0
---end scope---
myset1 SET 0
a INT 1
---end scope---
print (STRING)->VOID 0
printi (INT)->VOID 0
foo (INT)->INT 0
main ()->VOID 0
